app/repositories/ApprovalRepository.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import javax.inject.{Inject, Singleton}
+import models.errors.{DatabaseError, Errors, NotFoundError}
+import models.{ApprovalProcess, RequestOutcome}
+import play.api.Logger
+import play.api.libs.json.{Format, JsObject, Json}
+import play.modules.reactivemongo.ReactiveMongoComponent
+import repositories.formatters.ApprovalProcessFormatter
+import uk.gov.hmrc.mongo.ReactiveRepository
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+trait ApprovalRepository {
+  def update(id: String, process: JsObject): Future[RequestOutcome[String]]
+  def getById(id: String): Future[RequestOutcome[JsObject]]
+}
+
+@Singleton
+class ApprovalRepositoryImpl @Inject()(mongoComponent: ReactiveMongoComponent)
+  extends ReactiveRepository[ApprovalProcess, String](
+    collectionName = "approvalProcesses",
+    mongo = mongoComponent.mongoConnector.db,
+    domainFormat = ApprovalProcessFormatter.mongoFormat,
+    idFormat = implicitly[Format[String]]
+  ) with ApprovalRepository {
+
+  def update(id: String, process: JsObject): Future[RequestOutcome[String]] = {
+
+    Logger.info(s"Saving process $id to collection $collectionName")
```

Comment:
Apologies for this, but following this mornings discussion we should revert to the class specific logger.

Reply:
updated by latest commit